glslc/test/working_directory.py

```diff
@@ -148,15 +149,50 @@ class TestWorkDirEqInArg(expect.ValidNamedObjectFile):
     expected_object_filenames = ('a.spv',)
 
 
+class WorkDirAffectsNotLinkedFile(expect.SuccessfulReturn,
+                                  expect.CorrectObjectFilePreamble):
+
+    def check_output_in_subdir(self, status):
+        if hasattr(self, 'expected_object_filenames'):
+            input_filenames_to_verify = self.expected_object_filenames
+        else:
+            input_filenames_to_verify = status.input_filenames
+        for input_filename in input_filenames_to_verify:
+            object_filename = expect.get_object_filename(
+                os.path.basename(input_filename))
+            success, message = self.verify_object_file_preamble(
+                os.path.join(status.directory, 'subdir', object_filename))
+            if not success:
+                return False, message
+        return True, ''
+
+
 @inside_glslc_testsuite('WorkDir')
-class TestWorkDirCompileFileAbsolutePath(expect.ValidObjectFile):
+class TestWorkDirCompileFileAbsolutePath(WorkDirAffectsNotLinkedFile):
     """Tests -working-directory=<dir> when compiling input file with absolute
-    path."""
+    path. The output location should refer to the working directory"""
 
     shader = FileShader('void main() {}', '.vert')
+    environment = Directory('.', [
+        Directory('subdir', []),
+    ])
     glslc_args = ['-c', '-working-directory=subdir', shader]
```

Comment:
The previous test expect to find the output file in the same directory with the input shader file. But we should expect the output file in 'subdir'.